tests/scripts/all.sh

```diff
@@ -957,6 +949,20 @@ component_build_no_sockets () {
     make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0 -std=c99 -pedantic' lib
 }
 
+component_test_memory_buffer_allocator () {
+    msg "build: default config with memory buffer allocator enabled"
+    scripts/config.pl set MBEDTLS_MEMORY_BUFFER_ALLOC_C
+    scripts/config.pl set MBEDTLS_MEMORY_BACKTRACE
+    CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
```

Comment:
Does it make sense to run ASan with `MBEDTLS_MEMORY_BUFFER_ALLOC_C` also set?

We should have another test with a full config without `MBEDTLS_MEMORY_BUFFER_ALLOC_C` and ASan, to check for memory leaks in general.

Reply:
Regarding your first point: Yes, there's not much point in using ASan here - that's the whole point. I'll remove it.

Regarding your second point: Yes, that's another existing limitation worth fixing along the way.